tools/test-setup.sh

```diff
@@ -216,13 +216,14 @@ log notice "Installing $(python3 --version) venv and dependencies matching creat
 VIRTUAL_ENV=${VIRTUAL_ENV:-out/venvs/${HOSTNAME}}
 if [[ ! -d ${VIRTUAL_ENV} ]]; then
     log notice "Creating virtualenv ..."
-    python3 -m venv "${VIRTUAL_ENV}"
+    python3.11 -m venv "${VIRTUAL_ENV}"
```

Comment:
I do not fancy this option at all as it assumes you have a py311 interpreter. I might have py312 only and it will break the logic.

Reply:
The CI runs 3.11 and I would prefer to avoid any behavioural change base on the local set-up.